src/core/json-schema-components.jsx

```diff
@@ -50,8 +50,9 @@ export class JsonSchemaForm extends Component {
       schema = schema.toJS()
 
     let { type, format="" } = schema
-
-    let Comp = (format ? getComponent(`JsonSchema_${type}_${format}`) : getComponent(`JsonSchema_${type}`)) || getComponent("JsonSchema_string")
+    // In the json schema rendering code, we optimistically query our system for a number of components.
+    // If the component doesn't exist, we optionally suppress these warnings.
+    let Comp = (format ? getComponent(`JsonSchema_${type}_${format}`, false, { failSilently: true }) : getComponent(`JsonSchema_${type}`, false, { failSilently: true })) || getComponent("JsonSchema_string", false, { failSilently: true })
```

Comment:
**suggestion:** there's a _ton_ going on in L55 now, let's break out our common logic to a reusable function.

(let's follow the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))!)

```suggestion
    // If the component doesn't exist, we optionally suppress these warnings.
    let getComponentSilently = (name) => getComponent(name, false, { failSilently: true })
    let Comp = (format ? getComponentSilently(`JsonSchema_${type}_${format}`) : getComponentSilently(`JsonSchema_${type}`)) || getComponentSilently("JsonSchema_string")
```